workshops/util.py

```diff
@@ -471,3 +472,31 @@ def update_event_attendance_from_tasks(event):
 
 def universal_date_format(date):
     return '{:%Y-%m-%d}'.format(date)
+
+
+def get_members():
+    '''Get everyone who is a member of the Software Carpentry Foundation.'''
+
+    # Everyone who is an explicit member.
+    member_badge = Badge.objects.get(name='member')
+    queryset = member_badge.person_set.all()
+
+    # This is a really clumsy way to add everyone who qualifies by having taught recently.
+    today = datetime.date.today()
+    cutoff = datetime.date(year=today.year-2, month=today.month, day=today.day)
+    member_ids = set([p.id for p in queryset])
+    instructor_badge = Badge.objects.get(name='instructor')
+    instructors = instructor_badge.person_set.all()
+    instructor_role = Role.objects.get(name='instructor')
+    implicit_members = []
+    for i in instructors:
+        if i.id not in member_ids:
+            tasks = i.task_set.filter(role=instructor_role)
+            for t in tasks:
+                if t.event.start and (t.event.start >= cutoff):
+                    implicit_members.append(i)
+                    member_ids.add(i.id)
+                    break
```

Comment:
Okay, here are Django queries I came up with:

```python
def get_members():
    '''Get everyone who is a member of the Software Carpentry Foundation.'''

    today = datetime.date.today()
    cutoff = today - 2 * datetime.timedelta(days=365)

    member_badge = Badge.objects.get(name='member')
    instructor_badge = Badge.objects.get(name='instructor')
    instructor_role = Role.objects.get(name='instructor')

    # Everyone who is an explicit member.
    qs1 = Person.objects.filter(badges__in=[member_badge]).distinct()

    # Everyone who qualifies by having taught recently.
    qs2 = Person.objects.filter(task__role=instructor_role, task__event__start__gte=cutoff, badges__in=[instructor_badge]).distinct()

    queryset = qs1 | qs2
    return queryset
```

Now can someone validate that this yields correct results?